node/neighbor.go

```diff
@@ -179,17 +178,29 @@ func (localNode *LocalNode) splitNeighbors(filter func(*RemoteNode) bool) ([]*Re
 	return chordNeighbors, randomNeighbors
 }
 
-func (localNode *LocalNode) getSampledNeighbors(filter func(*RemoteNode) bool, chordNeighborSampleRate, randomNeighborSampleRate float64) []*RemoteNode {
+func (localNode *LocalNode) getSampledNeighbors(filter func(*RemoteNode) bool, chordNeighborSampleRate float64, chordNeighborMinSample int, randomNeighborSampleRate float64, randomNeighborMinSample int) []*RemoteNode {
 	var sampledNeighbors []*RemoteNode
 	chordNeighbors, randomNeighbors := localNode.splitNeighbors(filter)
 
-	numChordSamples := int(math.Ceil(chordNeighborSampleRate * float64(len(chordNeighbors))))
+	numChordSamples := int(chordNeighborSampleRate * float64(len(chordNeighbors)))
+	if numChordSamples < chordNeighborMinSample {
+		numChordSamples = chordNeighborMinSample
+	}
+	if numChordSamples > len(chordNeighbors) {
```

Comment:
Can `numChordSamples > len(chordNeighbors)` and `numChordSamples < chordNeighborMinSample` happen together?

Reply:
Yes, if `len(chordNeighbors) < chordNeighborMinSample `, both of these `if` conditions could be true